toolchain/check/decl_name_stack.cpp

```diff
@@ -245,6 +260,18 @@ auto DeclNameStack::TryResolveQualifier(NameContext& name_context,
             context_->classes().Get(class_decl->class_id).self_type_id);
         context_->NoteIncompleteClass(class_decl->class_id, builder);
         builder.Emit();
+      } else if (auto interface_decl = context_->insts()
+                                           .Get(name_context.resolved_inst_id)
+                                           .TryAs<SemIR::InterfaceDecl>()) {
+        CARBON_DIAGNOSTIC(
+            QualifiedDeclInUndefinedInterfaceScope, Error,
+            "Cannot declare a member of undefined interface `{0}`.",
```

Comment:
Why undefined here versus incomplete for the equivalent class diagnostic?

Reply:
For a class type, after `class C;` you can't write `let c: C`, but for an interface, after `interface I;` you can write `let T:! I;` -- an interface becomes complete after a declaration whereas a class only becomes complete after a definition. So "incomplete" wouldn't be correct terminology for an interface. We could use "undefined" in both cases, although I think "incomplete" is a more familiar term for classes coming from C++.

Reply:
It may be nice to find a consistent term... "undefined" might be better for consistency, but it has an awkward meaning to say a name is undefined.

Reply:
(I'm maybe less swayed by C++ here because I think it might help to give people the same term for when members can be defined for both)